vpr/src/route/connection_box_lookahead_map.cpp

```diff
@@ -438,46 +448,62 @@ static float get_connection_box_lookahead_map_cost(int from_node_ind,
     ssize_t dx = ssize_t(from_canonical_loc->first) - ssize_t(box_location.first);
     ssize_t dy = ssize_t(from_canonical_loc->second) - ssize_t(box_location.second);
 
-    int from_seg_index = g_cost_map.node_to_segment(from_node_ind);
-    Cost_Entry cost_entry = g_cost_map.find_cost(from_seg_index, box_id, dx, dy);
+    int from_seg_index = cost_map_.node_to_segment(from_node_ind);
+    Cost_Entry cost_entry = cost_map_.find_cost(from_seg_index, box_id, dx, dy);
+
+    if (!cost_entry.valid()) {
+        // there is no route
+#if 0
+        // useful for debugging but can be really noisy
+        VTR_LOG_WARN("Not connected %d (%s, %d) -> %d (%s, %d, (%d, %d))\n",
```

Comment:
Best to tie this to VTR_ENABLE_DEBUGGING with `VTR_LOGV_DEBUG` and `f_router_debug`, see https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/master/vpr/src/route/route_timing.cpp#L2046

Reply:
Done.